cache/src/main/java/org/apollo/cache/decoder/ObjectDefinitionDecoder.java

```diff
@@ -20,123 +19,193 @@
+				final int count = buffer.readUByte();
+				definition.setOriginalColours(new short[count]);
+				definition.setReplacementColours(new short[count]);
+				for (int i = 0; i < count; i++) {
+					definition.getOriginalColours()[i] = (short) buffer.readUShort();
+					definition.getReplacementColours()[i] = (short) buffer.readUShort();
+				}
+			} else if (opcode == 41) {
+				final int count = buffer.readUByte();
+				definition.setOriginalTextures(new short[count]);
+				definition.setReplacementTextures(new short[count]);
+				for (int i_7_ = 0; i_7_ < count; i_7_++) {
+					definition.getOriginalTextures()[i_7_] = (short) buffer.readUShort();
+					definition.getReplacementTextures()[i_7_] = (short) buffer.readUShort();
 				}
-			} else if (opcode == 60 || opcode >= 65 && opcode <= 68) {
-				data.getShort();
+			} else if (opcode == 62) {
+				definition.setMirrorModel(true);
+			} else if (opcode == 64) {
+				definition.setCastsShadow(false);
+			} else if (opcode == 65) {
+				definition.setScaleX(buffer.readUShort());
+			} else if (opcode == 66) {
+				definition.setScaleY(buffer.readUShort());
+			} else if (opcode == 67) {
+				definition.setScaleZ(buffer.readUShort());
+			} else if (opcode == 68) {
+				definition.setMapscene(buffer.readUShort());
 			} else if (opcode == 69) {
-				data.get();
-			} else if (opcode >= 70 && opcode <= 72) {
-				data.getShort();
+				definition.setRotationFlag(buffer.readUByte());
+			} else if (opcode == 70) {
+				definition.setTranslateX((buffer.readSignedShort()));
+			} else if (opcode == 71) {
+				definition.setTranslateY((buffer.readSignedShort()));
+			} else if (opcode == 72) {
+				definition.setTranslateZ((buffer.readSignedShort()));
 			} else if (opcode == 73) {
+				definition.setDecoration(true);
+			} else if (opcode == 74) {
 				definition.setObstructive(true);
 			} else if (opcode == 75) {
-				data.get();
-			} else if (opcode == 77) {
-				data.getShort();
-				data.getShort();
-				int count = data.get();
-				for (int i = 0; i <= count; i++) {
-					data.getShort();
+				definition.setHoldsItemPiles(buffer.readUByte());
+			} else if (opcode == 77 || opcode == 92) {
+				definition.setVarbit(buffer.readUShort());
+				if (65535 == definition.getVarbit()) {
+					definition.setVarbit(-1);
+				}
+				definition.setVarp(buffer.readUShort());
+				if (65535 == definition.getVarp()) {
+					definition.setVarp(-1);
+				}
+
+				int model = -1;
+				if (opcode == 92) {
+					model = buffer.readSmart32();
+				}
+
+				int length = buffer.readUByte();
+				definition.setMorphisms(new int[length + 2]);
+
+				for (int index = 0; index <= length; index++) {
+					definition.getMorphisms()[index] = buffer.readSmart32();
+				}
+
+				definition.getMorphisms()[1 + length] = model;
+			} else if (opcode == 78) {
+				definition.setF2117(buffer.readUShort());
+				definition.setF2118(buffer.readUByte());
+			} else if (opcode == 79) {
+				definition.setF2119(buffer.readUShort());
```

Comment:
If a field is unknown we don't assign it to unused fields, instead we just skip the data.

Reply:
To be faaaaaaaiiiiiiir I c&p this out of my cache editor. But i'll get rid of them.